font/src/ft/mod.rs

```diff
@@ -213,75 +222,50 @@ impl FreeTypeRasterizer {
         let size = Size::new(size.as_f32_pts() * self.device_pixel_ratio * 96. / 72.);
         self.pixel_size = f64::from(size.as_f32_pts());
 
+        let config = fc::Config::get_current();
+        let mut pattern = Pattern::new();
+        pattern.add_family(&desc.name);
+        pattern.add_pixelsize(self.pixel_size);
+        let hash = pattern.hash();
+
+        // Add style to a pattern
         match desc.style {
             Style::Description { slant, weight } => {
                 // Match nearest font
-                self.get_matching_face(&desc, slant, weight)
+                pattern.set_weight(weight.into_fontconfig_type());
+                pattern.set_slant(slant.into_fontconfig_type());
             },
             Style::Specific(ref style) => {
-                // If a name was specified, try and load specifically that font.
-                self.get_specific_face(&desc, &style)
+                // If a name was specified, try and load specifically that font
+                pattern.add_style(style);
             },
         }
-    }
-
-    fn full_metrics(&self, key: FontKey) -> Result<FullMetrics, Error> {
-        let face = self.faces.get(&key).ok_or(Error::FontNotLoaded)?;
 
-        let size_metrics = face.ft_face.size_metrics().ok_or(Error::MissingSizeMetrics)?;
-
-        let width = match face.ft_face.load_char('0' as usize, face.load_flags) {
-            Ok(_) => face.ft_face.glyph().metrics().horiAdvance / 64,
-            Err(_) => size_metrics.max_advance / 64,
-        } as f64;
-
-        Ok(FullMetrics { size_metrics, cell_width: width })
-    }
-
-    fn get_matching_face(
-        &mut self,
-        desc: &FontDesc,
-        slant: Slant,
-        weight: Weight,
-    ) -> Result<FontKey, Error> {
-        let mut pattern = Pattern::new();
-        pattern.add_family(&desc.name);
-        pattern.set_weight(weight.into_fontconfig_type());
-        pattern.set_slant(slant.into_fontconfig_type());
-        pattern.add_pixelsize(self.pixel_size);
-
-        self.query_font(pattern, desc)
-    }
-
-    fn get_specific_face(&mut self, desc: &FontDesc, style: &str) -> Result<FontKey, Error> {
-        let mut pattern = Pattern::new();
-        pattern.add_family(&desc.name);
-        pattern.add_style(style);
-        pattern.add_pixelsize(self.pixel_size);
-
-        self.query_font(pattern, desc)
-    }
-
-    fn query_font(&mut self, pattern: Pattern, desc: &FontDesc) -> Result<FontKey, Error> {
-        let config = fc::Config::get_current();
-        let fonts = fc::font_sort(&config, &mut pattern.clone())
+        // Match pattern and get a font list, use first font as a "primary" one and the rest for a
+        // font fallback
```

Comment:
The double line comment here just bothered me slightly, so here's a proposed fix:

```
        // Get font list using pattern; first font is the primary while the rest are fallback
```

That doesn't linebreak anymore and contains the exact same information.